src/js/lib/admin-page/components/Setting.js

```diff
@@ -0,0 +1,47 @@
+import Input from './Input';
+import Select from './Select';
+import { FormToggle } from '@wordpress/components';
+
+const Setting = ( { note, setting, onChange, label } ) => {
+	let input;
+	if ( typeof setting[ Object.keys( setting )[ 0 ] ] === 'string' ) {
+		input = (
+			<Input
+				className="text-input"
+				type="text"
+				name={ Object.keys( setting ) }
+				value={ setting[ Object.keys( setting )[ 0 ] ] }
+				onChange={ onChange }
+			/>
+		);
+	} else if ( typeof setting[ Object.keys( setting )[ 0 ] ] === 'boolean' ) {
+		input = (
+			<FormToggle
```

Comment:
Is there a reason you decided to use `FormToggle` vs. `FormControl`? The latter will give us a bit more in the way of accessibility via the [`help` prop](https://github.com/WordPress/gutenberg/blob/e24947929f31b34459a85d31b9c5372f50cdce01/packages/components/src/toggle-control/README.md#help).

Reply:
I was trying both of them, but `FormToggle` accepted a `name` prop that `ToggleControl` didn't, and the lack of `name` stopped my `onChange` function from updating state properly. Is there something I can do to get that to update state?

Reply:
digging into this a little bit deeper, the main reason I can't update state properly when using `ToggleControl` is because `event.target` is undefined when you try to toggle on/off with that. And the reason for that I think is because `ToggleControl` actually renders `FormToggle` under the hood, but doesn't allow the propagation of the event to the child component where `event.target` would actually be defined. Is there a workaround for that?

Reply:
Took a stab at reworking all the Settings to use the `Control` variants in #334. Let me know what you think!